hydra-task/src/main/java/com/addthis/hydra/task/map/MapFeeder.java

```diff
@@ -103,20 +103,23 @@ public MapFeeder(StreamMapper task, TaskDataSource source, int feeders) {
         }
 
         try {
+            boolean interrupt = false;
             if (source.isEnabled()) {
                 while (fillBuffer()) {
                     if (Thread.interrupted()) {
+                        interrupt = true;
                         closeSourceIfNeeded();
                     }
                 }
             }
+            final boolean interrupted = interrupt;
             closeSourceIfNeeded();
             joinProcessors();
             log.info("all ({}) task threads exited; sending taskComplete", feeders);
 
             // run in different threads to isolate them from interrupts. ie. "taskCompleteUninterruptibly"
-            // join awaits completion, is uninterruptible, and will propogate any exception
-            CompletableFuture.runAsync(task::taskComplete).join();
+            // join awaits completion, is uninterruptible, and will propagate any exception
+            CompletableFuture.runAsync(() -> task.taskComplete(interrupted)).join();
             // critical to get any file meta data written before process exits
             CompletableFuture.runAsync(MuxFileDirectoryCache::waitForWriteClosure).join();
```

Comment:
this looks like this can get called after another task in the pipeline has started, but this method acts on a static instance that may be being used by the new task. Aside from angry log messages about being unable to close things, I think it will eventually try to force close files / directories and that will probably cause errors for the next task.